jenkins/Jenkinsfile-release

```diff
@@ -0,0 +1,82 @@
+
+FLOW_VERSION = null
+DEFAULT_BRANCH = 'master'
+REGISTRY = 'docker.h2o.ai'
+IMAGE = "${REGISTRY}/opsh2oai/h2o-flow-build"
+
+try {
+    ansiColor('xterm') {
+        timestamps {
+            node('master') {
+                buildSummary.stageWithSummary('Checkout') {
+                    def final scmEnv = checkout scm
+                    echo "${scmEnv}"
+                    env.BRANCH_NAME = scmEnv['GIT_BRANCH'].replaceAll('origin/', '')
+
+                    if (env.BRANCH_NAME == DEFAULT_BRANCH) {
+                        FLOW_VERSION = "0.7.${currentBuild.number}"
+                    } else {
+                        FLOW_VERSION = "0.8.${currentBuild.number}-${env.BRANCH_NAME.replaceAll('/', '-')}-SNAPSHOT"
+                    }
+
+                    // setup custom DetailsSummary
+                    DetailsSummary detailsSummary = new DetailsSummary()
+                    detailsSummary.setEntry(this, 'Branch Name', env.BRANCH_NAME)
+                    detailsSummary.setEntry(this, 'SHA', scmEnv['GIT_COMMIT'])
+                    detailsSummary.setEntry(this, 'Version', FLOW_VERSION)
+                    buildSummary.get().addDetailsSummary(this, detailsSummary)
+                }
+
+                buildSummary.stageWithSummary("Build") {
+                    withCredentials([usernamePassword(credentialsId: REGISTRY, usernameVariable: 'REGISTRY_USERNAME', passwordVariable: 'REGISTRY_PASSWORD')]) {
+                        sh "docker login -u $REGISTRY_USERNAME -p $REGISTRY_PASSWORD ${REGISTRY}"
+                        sh "docker pull ${IMAGE}"
+                    }
+
+                    timeout(time: 5, unit: 'MINUTES') {
+                        docker.withRegistry("https://${REGISTRY}") {
+                            docker.image(IMAGE).inside("--init") {
+                                sh """
+                                    env
+                                    id
+                                    
+                                    ln -s /home/jenkins/flow-cache/node_modules/ ./node_modules
+                                    ln -s /home/jenkins/flow-cache/lib/ ./lib
+        
+                                    npm run build
+                                    sed -E "s/.+version.+/  \\"version\\": \\"$FLOW_VERSION\\",/" bower.json > bower.json.tmp && mv bower.json.tmp bower.json
+                                    sed -E "s/999\\.999\\.999/$FLOW_VERSION/" build/js/flow.js > build/js/flow.js.tmp && mv build/js/flow.js.tmp build/js/flow.js
+                                """
+                            }
+                        }
+                    }
+                }
+
+                buildSummary.stageWithSummary("Commit and Push") {
+                    sh """
+                        git add .
+                        git commit -m "Release $FLOW_VERSION"
+                        git push --set-upstream origin ${env.BRANCH_NAME}
```

Comment:
Do we want to allow releases from other branches than master?

Reply:
maybe for some debugging purposes? The only thing the release does is that it commits the built files and pushes tag. Anyway, we can disable builds of branches other than master inside Jenkins. I want to make sure, that we don't accidentally release a regular release from some other branch.